utils/python/CIME/build.py

```diff
@@ -301,6 +305,24 @@ def case_build(caseroot, case=None, testmode=False, sharedlib_only=False, model_
 
     if not model_only:
         logs = build_libraries(exeroot, caseroot, cimeroot, libroot, mpilib, lid, machines_file)
+        if sharedlib_only and comp_lnd == "clm" and not "clm4_0" in clm_config_opts:
```

Comment:
Yikes. Why are clm builds so strange? Why do we only build these if sharedlib_only? Why not put this code in the build_libraries routine?